src/Transactions/CreateTransaction/CreateTransactionContainer.js

```diff
@@ -121,39 +89,76 @@ export const CreateTransactionContainer = ({
       transactionType,
     };
 
+    return accumulatedData;
+  }, [allocationType, budget, budgetName, currency, fundId, funds, locale, transactionType]);
+
+  const handleErrorResponse = useCallback(async ({ formValues, errorResponse }) => {
+    const accumulatedData = getAccumulatedDataObject(formValues);
+
+    const message = await handleCreateTransactionErrorResponse({
+      ...accumulatedData,
+      errorResponse,
+      formValues,
+      transactionTypeKey,
+    });
+
+    showCallout({ message, type: 'error' });
+  }, [getAccumulatedDataObject, handleCreateTransactionErrorResponse, showCallout, transactionTypeKey]);
+
+  const saveTransactionStep = useCallback(async (formValues, { resultBudgetName }) => {
+    return batchTransactions({
+      transactionType: BATCH_TRANSACTION_TYPES.transactionsToCreate,
+      data: [{
+        ...omit(formValues, ['fundId']),
+        id: uuidv4(),
```

Comment:
Should this be within the UI's responsibility? cc @SerhiiNosko

Reply:
Currently, the BE prevents us from making a request without a unique `id`![Screenshot 2024-11-04 at 16 52 16](https://github.com/user-attachments/assets/7dd23e63-0404-4247-ab29-bf9ec0f6c0c0)